components/Comment/CommentList.tsx

```diff
@@ -0,0 +1,251 @@
+import { useCallback, useContext, useEffect, useMemo, useState } from "react";
+import { ID, parseUser, TopLevelDocument } from "~/config/types/root_types";
+import { Comment as CommentType, COMMENT_TYPES } from "./lib/types";
+import Comment from "./Comment";
+import { css, StyleSheet } from "aphrodite";
+import colors from "./lib/colors";
+import CommentEditor from "~/components/Comment/CommentEditor";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "~/redux";
+import findComment from "./lib/findComment";
+import { isEmpty } from "~/config/utils/nullchecks";
+import IconButton from "../Icons/IconButton";
+import { MessageActions } from "~/redux/message";
+import {
+  createCommentAPI,
+  updateCommentAPI,
+  fetchCommentsAPI,
+} from "./lib/api";
+import CommentPlaceholder from "./CommentPlaceholder";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faLongArrowDown } from "@fortawesome/pro-regular-svg-icons";
+import { CommentTreeContext } from "./lib/contexts";
+const { setMessage, showMessage } = MessageActions;
+
+type Args = {
+  parentComment?: CommentType;
+  comments?: Array<CommentType>;
+  document: TopLevelDocument;
+  isRootList?: boolean;
+  isFetchingList?: boolean;
+  totalCount: number;
+};
+
+const CommentList = ({
+  comments = [],
+  totalCount,
+  parentComment,
+  document,
+  isRootList = false,
+  isFetchingList = false,
+}: Args) => {
+  const [isFetchingMore, setIsFetchingMore] = useState<boolean>(false);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const currentUser = useSelector((state: RootState) =>
+    isEmpty(state.auth?.user) ? null : parseUser(state.auth.user)
+  );
+  const dispatch = useDispatch();
+  const commentTreeState = useContext(CommentTreeContext);
+
+  const fetchMore = async ({}) => {
+    setIsFetchingMore(true);
+    try {
+      const response = await fetchCommentsAPI({
+        documentId: document.id,
+        documentType: document.documentType,
+        sort: commentTreeState.sort,
+        filter: commentTreeState.filter,
+        page: currentPage,
```

Comment:
This feels like we'll be fetching a dupe page on the first load of fetchMore, but I might not be seeing it properly